### PR TITLE
Revert "feat: Update VHD scanning so it doesn't require RBAC admin"

### PR DESCRIPTION
**Pull request by babbageclunk**
...istrator role (#4711)"

This reverts commit 05ed64b01f41b2dbc3fbfde8d34f6eea9ce6ac8a.

It turns out that doing it this way doesn't work for trusted launch VMs, so we need to go back to the previous way and assign the RBAC Admin role to the identity for the sub (since it's done on a VM in a new RG).

**What type of PR is this?**
/kind regression

**What this PR does / why we need it**:
Reverts the change that removed the need for RBAC Admin role, because assigning the identity like that breaks scanning for Trusted Launch VMs.

**Which issue(s) this PR fixes**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

**Review by AlisonB319 — APPROVED** [review]

**Comment by cameronmeissner** [comment]
something seems off:

+ az vm create --resource-group vhd-scanning-1724179161-12170 --name vhd-scanning-vm-1724179160-8286 --image /subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/2204gen2containerd/versions/1.1724177255.26372 --vnet-name nodesig-pool-vnet-westus3 --subnet scanning --admin-username azureuser --admin-password ScanVM@1724179161 --os-disk-size-gb 50 --size Standard_D8ds_v5 --assign-identity '[system]'
WARNING: It is recommended to use parameter "--public-ip-sku Standard" to create new VM with Standard public IP. Please note that the default public IP used for VM creation will be changed from Basic to Standard in the future.
WARNING: No access was given yet to the 'vhd-scanning-vm-1724179160-8286', because '--scope' was not provided. You should setup by creating a role assignment, e.g. 'az role assignment create --assignee <principal-id> --role contributor -g vhd-scanning-1724179161-12170' would let it access the current resource group. To get the pricipal id, run 'az vm show -g vhd-scanning-1724179161-12170 -n vhd-scanning-vm-1724179160-8286 --query "identity.principalId" -otsv'
{
  "fqdns": "",
  "id": "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/vhd-scanning-1724179161-12170/providers/Microsoft.Compute/virtualMachines/vhd-scanning-vm-1724179160-8286",
  "identity": {
    "systemAssignedIdentity": "d2ee23c1-4a01-4128-81e8-7c0d4117fcac",
    "userAssignedIdentities": {}
  },
  "location": "westus3",
  "macAddress": "7C-1E-52-10-14-AE",
  "powerState": "VM running",
  "privateIpAddress": "10.0.0.4",
  "publicIpAddress": "20.168.86.176",
  "resourceGroup": "vhd-scanning-1724179161-12170",
  "zones": ""
}
./vhdbuilder/packer/vhd-scanning.sh: line 77: VM_NAME: unbound variable

File: vhdbuilder/packer/vhd-scanning.sh [hunk 1]
```diff
@@ -74,7 +72,10 @@ az vm create --resource-group $RESOURCE_GROUP_NAME \
     --admin-password $SCAN_VM_ADMIN_PASSWORD \
     --os-disk-size-gb 50 \
     ${VM_OPTIONS} \
-    --assign-identity "${SCANNING_MSI_RESOURCE_ID}"
+    --assign-identity "[system]"
+
+VM_PRINCIPLE_ID=$(az vm identity show --name $VM_NAME --resource-group $RESOURCE_GROUP_NAME --query principalId --output tsv)
```
Comment (cameronmeissner) [inline vhdbuilder/packer/vhd-scanning.sh:77]: we need to be setting VM_NAME

Comment (AlisonB319) [reply vhdbuilder/packer/vhd-scanning.sh:77]: I might have tweeked something the wrong way with a merge conflict

Comment (AlisonB319) [reply vhdbuilder/packer/vhd-scanning.sh]: jk looks like it should just be $SCAN_VM_NAME instead of $VM_NAME

**Pull request merged**